### PR TITLE
scx-git: init; scx: remove

### PR DESCRIPTION
**Pull request by JohnRTitor**
### :fish: What?

1. Remove scx to prevent conflicts.
2. Override the scx packages of Nixpkgs to provide a git version here

### :fishing_pole_and_fish: Why?

Scx is now available on Nixpkgs (stable version). So it makes sense to replace with the git version here.

### :fish_cake: Pending

- [x] Override the rust packages
- [x] Cook an update script for this, referencing [nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/scx/update.sh), or modify the existing one.
- [ ] Publishing to news' channel.

After this is done, we will need to announce to the users asking to change their packages to `scx_git.full`. If they want to use the nixpkgs version, they should switch to `scx.full`.

### :whale: Extras

I will recommend merging this after these PR reach unstable.

- https://nixpk.gs/pr-tracker.html?pr=352077 - rusty init
- https://nixpk.gs/pr-tracker.html?pr=352811 - csheds -> cscheds

### CODE REVIEW

File: overlays/default.nix [hunk 1]
```diff
@@ -251,17 +251,7 @@ in
...
-    scx-bpfland = final.callPackage ../pkgs/scx/bpfland { inherit scx-common; };
-    scx-lavd = final.callPackage ../pkgs/scx/lavd { inherit scx-common; };
-    scx-layered = final.callPackage ../pkgs/scx/layered { inherit scx-common; };
-    scx-mitosis = final.callPackage ../pkgs/scx/mitosis { inherit scx-common; };
-    scx-rlfifo = final.callPackage ../pkgs/scx/rlfifo { inherit scx-common; };
-    scx-rustland = final.callPackage ../pkgs/scx/rustland { inherit scx-common; };
-    scx-rusty = final.callPackage ../pkgs/scx/rusty { inherit scx-common; };
-    scx-stats = final.callPackage ../pkgs/scx/stats { inherit scx-common; };
-  };
+  scx-git = callOverride ../pkgs/scx-git { };
```
Comment (PedroHLC) [inline overlays/default.nix:254]:
```suggestion
  scx_git = callOverride ../pkgs/scx-git { };
```

**Comment by JohnRTitor** [comment]
```
❯  nix build .#scx_git.full .#scx_git.rustland .#scx_git.bpfland .#scx_git.cscheds .#scx_git.lavd .#scx_git.layered .#scx_git.rlfifo --print-out-paths 

/nix/store/ld2yx48yviiz0lpqac991l7xnsnmfgha-scx_full-unstable-20241101144355-9bba51a
/nix/store/xc9w481g4li7fwkyj317acgr462xfkdq-scx_rustland-unstable-20241101144355-9bba51a
/nix/store/xx68snlgqrspb3j298wc2hsqz3gkwkhx-scx_bpfland-unstable-20241101144355-9bba51a
/nix/store/q4l2qpd1nfyizhds9ax19805s85h2q0r-scx_csheds-unstable-20241101144355-9bba51a
/nix/store/ybs3n39sl9bw7swv7kg5nfp79kfvfg3k-scx_lavd-unstable-20241101144355-9bba51a
/nix/store/0lry6f5kknh16rfjmfbsgrbvdk522fp6-scx_layered-unstable-20241101144355-9bba51a
/nix/store/c1yqi50485ga1lj1in6gv06cy8jvxprm-scx_rlfifo-unstable-20241101144355-9bba51a
```

Everything above builds at the moment, and technically this is ready for merging. But I would like to wait until some merged PRs on Nixpkgs is available on unstable.

File: pkgs/scx-git/full.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+final.scx_git.cscheds.overrideAttrs (oldAttrs: {
+  pname = "scx_full";
+  postInstall =
+    (oldAttrs.postInstall or "")
+    + ''
+      cp ${lib.getExe final.scx_git.bpfland} $out/bin/
+      cp ${lib.getExe final.scx_git.lavd} $out/bin/
+      cp ${lib.getExe final.scx_git.layered} $out/bin/
+      cp ${lib.getExe final.scx_git.rlfifo} $out/bin/
+      cp ${lib.getExe final.scx_git.rustland} $out/bin/
```
Comment (JohnRTitor) [inline pkgs/scx-git/full.nix:14]:
```suggestion
      cp ${lib.getExe final.scx_git.rustland} $out/bin/
      cp ${lib.getExe final.scx_git.rusty} $out/bin/

```

File: overlays/default.nix [hunk 3]
```diff
@@ -251,16 +251,17 @@ in
...
+  scx_git = {
+    bpfland = final.callPackage ../pkgs/scx-git/bpfland { inherit scx-common; };
+    cscheds = final.callPackage ../pkgs/scx-git/cscheds.nix { inherit scx-common; };
+    lavd = final.callPackage ../pkgs/scx-git/lavd { inherit scx-common; };
+    layered = final.callPackage ../pkgs/scx-git/layered { inherit scx-common; };
+    rlfifo = final.callPackage ../pkgs/scx-git/rlfifo { inherit scx-common; };
+    rustland = final.callPackage ../pkgs/scx-git/rustland { inherit scx-common; };
+    # FIXME-SCX: rusty will fail to build unless https://github.com/NixOS/nixpkgs/pull/352077 is available in unstable
+    # rusty = final.callPackage ../pkgs/scx-git/rusty { inherit scx-common; };
+    # FIXME-SCX: add rusty to full once the above PR is available in unstable
```
Comment (JohnRTitor) [inline overlays/default.nix:263]:
```suggestion
    rusty = final.callPackage ../pkgs/scx-git/rusty { inherit scx-common; };
```

File: pkgs/scx-git/cscheds.nix [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{ scx
+, scx-common
+,
+}:
+
+# FIXME-SCX: change csheds to cscheds
+# once https://github.com/NixOS/nixpkgs/pull/352811 is available in unstable
+scx.csheds.overrideAttrs (oldAttrs: {
```
Comment (JohnRTitor) [inline pkgs/scx-git/cscheds.nix:8]:
```suggestion
scx.cscheds.overrideAttrs (oldAttrs: {
```

**Review by JohnRTitor — COMMENTED** [review]
TODO changes to make:

**Comment by JohnRTitor** [comment]
Work is pretty much done ^^ except for the above changes which depend on nixpkgs bump.

**Comment by JohnRTitor** [comment]
If this does not work I am done with fixing the CI.

I have no clue where the problem lies my gosh.

**Comment by JohnRTitor** [comment]
@PedroHLC can we get this reviewed :)

**Comment by DataEraserC** [comment]
Hello maintainers, just a reminder to inform users about updating their configurations to use scx_git.full instead of the default pkgs.scx. It's also recommended to set a correct default value that aligns with scx_git.full, as the previous default (pkgs.scx) may lead to unexpected issues.

**Comment by JohnRTitor** [comment]
> It's also recommended to set a correct default value that aligns with scx_git.full

Yup, PRs welcomed. 
We need to change https://github.com/chaotic-cx/nyx/blob/09f0dcfa6a8932850f2291717a271598d3098aad/modules/nixos/scx.nix#L14 to 
https://github.com/NixOS/nixpkgs/blob/1f6ff249fe86a678965436f4d3c9c4a7e5601562/nixos/modules/services/scheduling/scx.nix#L23-L36

And we can remove the module entirely from here once the upstream PR is merged.

**Pull request merged**